src/Umbraco.Core/IO/IOHelper.cs

```diff
@@ -188,5 +188,62 @@ public string GetRelativePath(string path)
             return PathUtility.EnsurePathIsApplicationRootPrefixed(path);
         }
 
+        /// <summary>
+        /// Retrieves array of temporary folders from the hosting environment.
+        /// </summary>
+        /// <returns>Array of <see cref="DirectoryInfo"/> instances.</returns>
+        public DirectoryInfo[] GetTempFolders()
+        {
+            var tempFolderPaths = new[]
+            {
+                _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempFileUploads)
+            };
+
+            foreach (var tempFolderPath in tempFolderPaths)
+            {
+                // Ensure it exists
+                Directory.CreateDirectory(tempFolderPath);
+            }
+
+            return tempFolderPaths.Select(x => new DirectoryInfo(x)).ToArray();
+        }
```

Comment:
Not sure why this returns an array, do we every have multiple temp folders?

Reply:
Not sure either, but this is just copied over from existing code (I've moved it into the `IOHelper` from the original scheduled task).